Source/WebKit/NetworkProcess/storage/LocalStorageManager.cpp

```diff
@@ -107,40 +107,44 @@ bool LocalStorageManager::isActive() const
 
 bool LocalStorageManager::hasDataInMemory() const
 {
-    bool hasDataInLocalStorageArea = is<MemoryStorageArea>(m_localStorageArea) && !m_localStorageArea->isEmpty();
-    return hasDataInLocalStorageArea || (m_transientStorageArea && !m_transientStorageArea->isEmpty());
+    RefPtr memoryLocalStorage = dynamicDowncast<MemoryStorageArea>(m_localStorageArea);
+    if (memoryLocalStorage && !memoryLocalStorage->isEmpty())
+        return true;
+
+    RefPtr transientStorageArea = m_transientStorageArea;
+    return transientStorageArea && !transientStorageArea->isEmpty();
 }
 
 void LocalStorageManager::clearDataInMemory()
 {
-    if (is<MemoryStorageArea>(m_localStorageArea))
-        m_localStorageArea->clear();
+    if (RefPtr storage = dynamicDowncast<MemoryStorageArea>(m_localStorageArea))
+        storage->clear();
 
-    if (m_transientStorageArea)
-        m_transientStorageArea->clear();
+    if (RefPtr transientStorageArea = m_transientStorageArea)
```

Comment:
To match the other local variable names, this would just be `transientStorage`. Shorter is likely better, but also more consistent would be nicer. Not really important either way. If I was writhing the code, in a class named "storage manager" I’d use the word "area" rather than "storage", since "storage" is implied by the class.